client/client.go

```diff
@@ -100,18 +100,18 @@ type ClientStatsReporter interface {
 }
 
 type AllocRunner interface {
```

Comment:
This seems like a pretty central interface that should be well commented

Reply:
but totally outside the scope of this PR just brain dumping

Reply:
Yeah... it's currently unused.... in theory Client tests could use a mock AllocRunner, but the plumbing isn't even setup for that. I'll add a comment.